src/query/sql/src/planner/binder/project.rs

```diff
@@ -150,27 +152,143 @@ impl<'a> Binder {
         let mut output = SelectList::<'a>::default();
         for select_target in select_list {
             match select_target {
-                SelectTarget::QualifiedName(names) => {
+                SelectTarget::QualifiedName {
+                    qualified: names,
+                    exclude,
+                } => {
                     // Handle qualified name as select target
+                    let mut exclude_cols: HashSet<String> = HashSet::new();
+                    if let Some(cols) = exclude {
+                        for col in cols {
+                            exclude_cols.insert(col.name.clone());
+                        }
+                        if exclude_cols.len() < cols.len() {
+                            // * except (id, id)
+                            return Err(ErrorCode::SemanticError("duplicate column name"));
+                        }
+                    }
+
+                    // Pre-check exclude_col is legal
+                    let precheck_exclude_cols = |input_context: &BindContext,
+                                                 exclude_cols: &HashSet<String>,
+                                                 db_name: Option<&Identifier>,
+                                                 table_name: Option<&Identifier>|
+                     -> Result<()> {
+                        let all_columns_bind = input_context.all_column_bindings();
+                        let mut qualified_cols_name: HashMap<String, u8> = HashMap::new();
+                        for i in all_columns_bind {
+                            if let (None, None) = (db_name, table_name) {
```

Comment:
I recommend to use `match (db_name, table_name) {}`

Reply:
I use this match first. But I think this arm is unnecessary.

```
match  (db_name, table_name) {
   (Some(db_name), None ) => {}
}
```

So I use `if let`.

Reply:
> But I think this arm is unnecessary.

Although there is an unnecessary arm, they are more readable than three `if let` cases.

Reply:
> > But I think this arm is unnecessary.
> 
> Although there is an unnecessary arm, they are more readable than three `if let` cases.

Agreed, it's better in terms of readability. Let's change it.

Reply:
> I recommend to use `match (db_name, table_name) {}`

Done